sparkmagic/sparkmagic/tests/test_handlers.py

```diff
@@ -62,7 +64,7 @@ def setUp(self):
 
         # Mock request
         self.request = MagicMock()
-        self.request.body = json.dumps({"path": self.path, "username": self.username, "password": self.password, "endpoint": self.endpoint})
+        self.request.body = json.dumps({"path": self.path, "username": self.username, "password": self.password, "endpoint": self.endpoint, "auth_type": self.auth_type})
```

Comment:
There's no tests here to test for when `auth` is not in the payload at all. There should be two tests for that case:

1. `auth` is not on payload but due to empty username and password, auth should default to None.
1. `auth` is not on payload but due to non-empty username and password, auth should default to Basic Auth.